File: app/src/main/java/com/past/music/entity/ArtistEntity.java
package com.past.music.entity;

import android.os.Parcel;
import android.os.Parcelable;

/**
 * Created by gaojin on 2017/3/11.
 */

public class ArtistEntity implements Parcelable {

    public String artist_name;
    public int number_of_tracks;
    public long artist_id;
    public String artist_sort;

    public ArtistEntity() {
    }

    public ArtistEntity(Parcel in) {
        artist_name = in.readString();
        number_of_tracks = in.readInt();
        artist_id = in.readLong();
        artist_sort = in.readString();
    }

    public static final Creator<ArtistEntity> CREATOR = new Creator<ArtistEntity>() {
        @Override
        public ArtistEntity createFromParcel(Parcel in) {
            return new ArtistEntity(in);
        }

        @Override
        public ArtistEntity[] newArray(int size) {
            return new ArtistEntity[size];
        }
    };

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(artist_name);
        dest.writeInt(number_of_tracks);
        dest.writeLong(artist_id);
        dest.writeString(artist_sort);
    }

    public String getArtist_name() {
        return artist_name;
    }

    public void setArtist_name(String artist_name) {
        this.artist_name = artist_name;
    }

    public int getNumber_of_tracks() {
        return number_of_tracks;
    }

    public void setNumber_of_tracks(int number_of_tracks) {
        this.number_of_tracks = number_of_tracks;
    }

    public long getArtist_id() {
        return artist_id;
    }

    public void setArtist_id(long artist_id) {
        this.artist_id = artist_id;
    }

    public String getArtist_sort() {
        return artist_sort;
    }

    public void setArtist_sort(String artist_sort) {
        this.artist_sort = artist_sort;
    }
}

File: app/src/main/java/com/se/music/common/entity/ArtistEntity.kt
package com.se.music.common.entity

import android.os.Parcel
import android.os.Parcelable

/**
 *Author: gaojin
 *Time: 2018/5/13 下午4:24
 */

class ArtistEntity() : Parcelable {
    var artistName: String? = null
    var numberOfTracks: Int = 0
    var artistId: Long = 0
    var artistSort: String? = null

    constructor(parcel: Parcel) : this() {
        artistName = parcel.readString()
        numberOfTracks = parcel.readInt()
        artistId = parcel.readLong()
        artistSort = parcel.readString()
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(artistName)
        parcel.writeInt(numberOfTracks)
        parcel.writeLong(artistId)
        parcel.writeString(artistSort)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<ArtistEntity> {
        override fun createFromParcel(parcel: Parcel): ArtistEntity {
            return ArtistEntity(parcel)
        }

        override fun newArray(size: Int): Array<ArtistEntity?> {
            return arrayOfNulls(size)
        }
    }
}